rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -287,14 +292,30 @@ extern "C"
             return NULL;
         }
 
+        rmw_guard_condition_t * graph_guard_condition = rmw_create_guard_condition();
+        if (!graph_guard_condition) {
+            // error already set
+            return NULL;
+        }
+
+        FastRTPSNodeImpl * node_impl = nullptr;
+        try {
+            node_impl = new FastRTPSNodeImpl();
+        } catch(std::bad_alloc & exc) {
+            RMW_SET_ERROR_MSG("failed to allocate node impl struct");
```

Comment:
Yes. The function was not setup to handle unrolling. None of the functions in this file are except one. So I didn't want to change that in these changes. In general I think this time implementation needs some clean up.